src/ripple/app/misc/impl/Transaction.cpp

```diff
@@ -167,16 +167,25 @@ Transaction::load(
 Json::Value
 Transaction::getJson(JsonOptions options, bool binary) const
 {
-    Json::Value ret(mTransaction->getJson(JsonOptions::none, binary));
+    // Note, we explicitly suppress `include_date` option here
+    Json::Value ret(
+        mTransaction->getJson(options % JsonOptions::include_date, binary));
 
-    if (mInLedger)
+    if (mLedgerIndex)
     {
-        ret[jss::inLedger] = mInLedger;  // Deprecated.
-        ret[jss::ledger_index] = mInLedger;
+        if (!(options & JsonOptions::disable_API_prior_V2))
+        {
+            // Behaviour before API version 2
+            ret[jss::inLedger] = mLedgerIndex;
+        }
+
+        // TODO: disable_API_prior_V3 to disable output of both `date` and
+        // `ledger_index` elements (taking precedence over include_date)
+        ret[jss::ledger_index] = mLedgerIndex;
```

Comment:
No, this must be enabled in API version 1, but can be _also_ present in version 2. Since in API version 1 there's no alternative, I'd rather keep in version 2 as well, to make transition to version 2 easier. It's enough that folks need to deal with `hash` field changing location in this release.

Reply:
I see, I thought it needs to be removed from the `tx_json` field.